projects/pastanaga/src/lib/calendar/date-input.component.ts

```diff
@@ -0,0 +1,102 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, Input, Output, Renderer2 } from '@angular/core';
+import { PopupComponent } from '../popup/popup.component';
+import { PopupService } from '../popup/popup.service';
+import { markForCheck } from '../common/utils';
+
+@Component({
+    selector: 'pa-date-input',
+    templateUrl: './date-input.component.html',
+    styleUrls: ['../popup/_popup.scss', './date-input.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateInputComponent extends PopupComponent {
```

Comment:
why are you extending `PopupComponent`? I think it would make more sense to extend `TextFieldCommon` so you can inherit all the properties like `accent`, `lessen`...